### PR TITLE
REGRESSION(252640@main): [ macOS iOS15 ] TestWebKitAPI.WebKit.AlternativeServicesDefaultDirectoryCreation is a constant failure

### PR DESCRIPTION
**Pull request by szewai**
#### 3da73044142a7040e9d068489499450e86ce0399
<pre>
REGRESSION(252640@main): [ macOS iOS15 ] TestWebKitAPI.WebKit.AlternativeServicesDefaultDirectoryCreation is a constant failure
<a href="https://bugs.webkit.org/show_bug.cgi?id=242951">https://bugs.webkit.org/show_bug.cgi?id=242951</a>

Reviewed by Chris Dumez.

The path included in the message (i.e. the path that is actually in use) is the real path of temporary folder (i.e.
&quot;/private/var/tmp&quot;, because we resolve paths with realpath() to create sandbox extension), so let&apos;s look for the real
path in the message.

* Tools/TestWebKitAPI/Tests/WebKitCocoa/ServiceWorkerBasic.mm:

Canonical link: <a href="https://commits.webkit.org/252681@main">https://commits.webkit.org/252681@main</a>
</pre>

### CODE REVIEW

**Comment by cdumez** [comment]
Why don't we `realPath("/var/tmp")` in the test? Your approach wouldn't work if realPath("/var/tmp") doesn't contain /var/tmp

**Comment by szewai** [comment]
> Why don't we `realPath("/var/tmp")` in the test? Your approach wouldn't work if realPath("/var/tmp") doesn't contain /var/tmp

Yeah we can also compare with the real path. Will update

File: Tools/TestWebKitAPI/Tests/WebKitCocoa/ServiceWorkerBasic.mm [hunk 1]
```diff
@@ -1441,7 +1443,10 @@ - (void)userContentController:(WKUserContentController *)userContentController d
     [webView loadRequest:server.request("/second.html"_s)];
     TestWebKitAPI::Util::run(&done);
     done = false;
-    EXPECT_TRUE(retrievedString.contains("\"path\": \"/var/tmp\""_s));
+    String tempDirectoryString = tempDirectory;
+    NSString* tempDirectoryInUse = FileSystem::realPath(tempDirectoryString);
```
Comment (cdumez) [inline Tools/TestWebKitAPI/Tests/WebKitCocoa/ServiceWorkerBasic.mm]: nit: We can probably call `FileSystem::realPath(tempDirectory)` and get rid of the `tempDirectoryString` local.

**Review by cdumez — APPROVED** [review]
LGTM

**Pull request merged**